dm_control/locomotion/walkers/assets/dog_v2/add_muscles.py

```diff
@@ -0,0 +1,435 @@
+
+wrap_geoms_legs = ['shoulder_L_wrapping', 'elbow_L_wrapping',
+                   'wrist_L_wrapping', 'finger_L_wrapping', 'hip_L_wrapping',
+                   'knee_L_wrapping', 'toe_L_wrapping',
+
+                   'shoulder_R_wrapping', 'elbow_R_wrapping',
+                   'wrist_R_wrapping', 'finger_R_wrapping', 'hip_R_wrapping',
+                   'knee_R_wrapping', 'toe_R_wrapping']
+
+
+def getClosestGeom(kd_tree, point, mjcf, mtu):
+  """
+    Finds the closest geometric body to a given point using a KD-tree.
+
+    Args:
+        kd_tree (dict): A dictionary containing KD-trees for different bones.
+        point (array-like): The coordinates of the target point.
+        mjcf: The MJCF (MuJoCo XML) object.
+        mtu: The muscle-tendon unit (MTU) identifier.
+
+    Returns:
+        closest_geom_body: The closest geometric body (as a Mujoco body object)
+        to the target point.
+  """
+  dist = 100000000000
+  closest_geom_body = 0
+  for bone, tree in kd_tree.items():
+    dist_new, i = tree.query(np.array([point]), k=1)
+    geom = mjcf.find("geom", bone)
+    body = geom.parent
+
+    if mtu in muscle_legs:
+      if body.name not in ["upper_leg_L",
+                           "upper_leg_R", "lower_leg_L", "lower_leg_R",
+                           "foot_L", "foot_R", "toe_L", "toe_R",
+                           "scapula_L", "scapula_R", "upper_arm_L",
+                           "upper_arm_R", "lower_arm_L", "lower_arm_R",
+                           "hand_L", "hand_R", "finger_L", "finger_R",
+                           "pelvis"]:
+        continue
+    else:
+      if body.name in ["upper_leg_L",
+                       "upper_leg_R", "lower_leg_L", "lower_leg_R",
+                       "foot_L", "foot_R", "toe_L", "toe_R",
+                       "lower_arm_L", "lower_arm_R",
+                       "hand_L", "hand_R", "finger_L", "finger_R"]:
+        continue
+
+    if dist_new < dist:
+      dist = dist_new
+      closest_geom_body = body
+
+  return closest_geom_body
+
+
+def calculate_transformation(element):
+  """
+    Calculates the transformation matrix from the root to the given element.
+
+    This function iterates through the parent elements of the given element and
+    calculates the transformation matrices from the root until reaching the
+    given element. It considers the position and orientation (quaternion)
+    attributes of each element to construct the transformation matrices.
+    The final transformation matrix represents the cumulative transformation
+    from the root to the given element.
+
+    Args:
+        element: The MuJoCo element for which the transformation matrix
+          is calculated.
+
+    Returns:
+        T: The transformation matrix from the root to the given element.
+  """
+  # Calculate all transformation matrices from root until this element
+  all_transformations = []
+
+  while type(element) == mjcf_module.element._ElementImpl:
```

Comment:
nit: Generally you should use internals like this because they are not guaranteed to stay the same. Probably not super important in this case but you could try checking whether it's an instance of `mjcf.Element` instead?